frontend/src/components/phones/onboarding/RealPhoneSetup.tsx

```diff
@@ -0,0 +1,318 @@
+  const phoneWithMostRecentlySentVerificationCode =
+    getPhoneWithMostRecentlySentVerificationCode(
+      props.phonesPendingVerification
+    );
+  const [verificationCode, setVerificationCode] = useState("");
+  const verificationSentDate = parseDate(
+    phoneWithMostRecentlySentVerificationCode.verification_sent_date
+  );
+  const [remainingTime, setRemainingTime] = useState(
+    getRemainingTime(verificationSentDate)
+  );
+
+  const onChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+    setVerificationCode(event.target.value);
+  };
+
+  const onSubmit: FormEventHandler = async (event) => {
+    event.preventDefault();
+
+    const response = await props.submitPhoneVerification(
+      phoneWithMostRecentlySentVerificationCode.id,
+      {
+        number: phoneWithMostRecentlySentVerificationCode.number,
+        verification_code: verificationCode,
+      }
+    );
+
+    if (response.ok) {
+      leadRef.current?.classList.add(`${styles["is-success"]}`);
+    } else {
+      leadRef.current?.classList.add(`${styles["is-error"]}`);
+      inputRef.current?.classList.add(`${styles["is-error"]}`);
+    }
+  };
+
+  useInterval(() => {
+    if (remainingTime < 0) {
+      expiredErrorMsg.current?.classList.remove(`${styles["is-hidden"]}`);
+      formRef.current?.classList.add(`${styles["is-hidden"]}`);
+    }
+
+    setRemainingTime(getRemainingTime(verificationSentDate));
+  }, 1000);
+
+  const errorTimeExpired = (
+    <div
+      ref={expiredErrorMsg}
+      className={`${styles["step-input-verificiation-code-timeout"]} ${styles["is-hidden"]} `}
+    >
+      <p>{l10n.getString("phone-onboarding-step3-error-exipred")}</p>
+      <Button className={styles.button} type="submit">
+        {l10n.getString("phone-onboarding-step3-error-cta")}
+      </Button>
+    </div>
+  );
+
+  const codeEntryPlaceholder = "000000";
+
+  return (
+    <div
+      className={`${styles.step}  ${styles["step-input-verificiation-code"]} `}
+    >
+      <div ref={leadRef} className={styles.lead}>
+        <div
+          className={`${styles["step-input-verificiation-code-lead-default"]}`}
+        >
+          {/* Default state */}
+          <img src={EnteryVerifyCode.src} alt="" width={300} />
+          <h2>{l10n.getString("phone-onboarding-step2-headline")}</h2>
+        </div>
+        <div
+          className={`${styles["step-input-verificiation-code-lead-error"]} `}
+        >
+          {/* Timeout error state */}
+          <img src={EnteryVerifyCodeError.src} alt="" width={170} />
+          <h2 className={`${styles["is-error"]} `}>
+            {l10n.getString("phone-onboarding-step3-code-fail-title")}
+          </h2>
+          <p>{l10n.getString("phone-onboarding-step3-code-fail-body")}</p>
+        </div>
+      </div>
+      {/* Add error class of `mzp-is-error` */}
+
+      {/* TODO: Add logic to display timeout error */}
+      {errorTimeExpired}
+
+      {/* <form onSubmit={onSubmit} className={`${styles.form}`}> */}
```

Comment:
sugguestion (non-blocking): Remove commented code?

Reply:
Removed in de725280b1a1fa9cb499491b2078e30f3e43dcd1